#!/usr/bin/env python
PACKAGE='changer'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#list of parameter
gen.add('cob_x_offset_m',double_t,0,"h",0,-1,1)
gen.add('cob_y_offset_m',double_t,0,"h",0,-1,1)
gen.add('hip_roll_swap_angle_rad',double_t,0,"h",0,-1,1)
gen.add('foot_roll_gyro_p_gain',double_t,0,"h",0,-10,10)
gen.add('foot_roll_gyro_d_gain',double_t,0,"h",0,-10,10)
gen.add('foot_pitch_gyro_p_gain',double_t,0,"h",0,-10,10)
gen.add('foot_pitch_gyro_d_gain',double_t,0,"h",0,-10,10)
gen.add('foot_roll_angle_p_gain',double_t,0,"h",0,-10,10)
gen.add('foot_roll_angle_d_gain',double_t,0,"h",0,-10,10)
gen.add('foot_pitch_angle_p_gain',double_t,0,"h",0,-10,10)
gen.add('foot_pitch_angle_d_gain',double_t,0,"h",0,-10,10)
gen.add('foot_x_force_p_gain',double_t,0,"h",0,-1,1)
gen.add('foot_x_force_d_gain',double_t,0,"h",0,-1,1)
gen.add('foot_y_force_p_gain',double_t,0,"h",0,-1,1)
gen.add('foot_y_force_d_gain',double_t,0,"h",0,-1,1)
gen.add('foot_z_force_p_gain',double_t,0,"h",0,-1,1)
gen.add('foot_z_force_d_gain',double_t,0,"h",0,-1,1)
gen.add('foot_roll_torque_p_gain',double_t,0,"h",0,-1,1)
gen.add('foot_roll_torque_d_gain',double_t,0,"h",0,-1,1)
gen.add('foot_pitch_torque_p_gain',double_t,0,"h",0,-1,1)
gen.add('foot_pitch_torque_d_gain',double_t,0,"h",0,-1,1)

gen.add('roll_gyro_cut_off_frequency',double_t,0,"h",0,0,500)
gen.add('pitch_gyro_cut_off_frequency',double_t,0,"h",0,0,500)
gen.add('roll_angle_cut_off_frequency',double_t,0,"h",0,0,500)
gen.add('pitch_angle_cut_off_frequency',double_t,0,"h",0,0,500)
gen.add('foot_x_force_cut_off_frequency',double_t,0,"h",0,0,500)
gen.add('foot_y_force_cut_off_frequency',double_t,0,"h",0,0,500)
gen.add('foot_z_force_cut_off_frequency',double_t,0,"h",0,0,500)
gen.add('foot_roll_torque_cut_off_frequency',double_t,0,"h",0,0,500)
gen.add('foot_pitch_torque_cut_off_frequency',double_t,0,"h",0,0,500)


exit(gen.generate(PACKAGE,"changer","MyParam"))